### PR TITLE
[Merged by Bors] - chore: forward-port leanprover-community/mathlib#18429

### PR DESCRIPTION
**Pull request by eric-wieser**
Also removes a duplicate lemma that was added by accident while porting.

---

- [x] depends on: #2553

[Open in Gitpod](https://gitpod.io/from-referrer/)

* [`data.multiset.pi`@`9003f28797c0664a49e4179487267c494477d853`..`4c586d291f189eecb9d00581aeb3dd998ac34442`](https://leanprover-community.github.io/mathlib-port-status/file/data/multiset/pi?range=9003f28797c0664a49e4179487267c494477d853..4c586d291f189eecb9d00581aeb3dd998ac34442)
* [`data.finset.pi`@`9003f28797c0664a49e4179487267c494477d853`..`4c586d291f189eecb9d00581aeb3dd998ac34442`](https://leanprover-community.github.io/mathlib-port-status/file/data/finset/pi?range=9003f28797c0664a49e4179487267c494477d853..4c586d291f189eecb9d00581aeb3dd998ac34442)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2553
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2553~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/Multiset/Pi.lean [hunk 1]
```diff
@@ -139,14 +140,11 @@ theorem pi.cons_ext {m : Multiset α} {a : α} (f : ∀ a' ∈ a ::ₘ m, δ a')
   · rw [Pi.cons_ne _ h]
 #align multiset.pi.cons_ext Multiset.pi.cons_ext
 
-theorem pi.con_ext {m : Multiset α} {a : α} (f : ∀ a' ∈ a ::ₘ m, δ a') :
-    (Pi.cons m a (f _ (mem_cons_self _ _)) fun a' ha' => f a' (mem_cons_of_mem ha')) = f := by simp
-
```
Comment (eric-wieser) [inline Mathlib/Data/Multiset/Pi.lean:144]: `pi.cons_ext` exists immediately above with the same statement; I would guess this was an accidentally-committed test.

**Comment by RemyDegenne** [comment]
Thanks!
bors merge

**Pull request closed without merging**